ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx

```diff
@@ -117,6 +117,21 @@ export const PodsLogsViewer = (props: PodLogsProps & {fullscreen?: boolean}) =>
                             }}>
                             {prefs.appDetails.darkMode ? <i className='fa fa-sun' /> : <i className='fa fa-moon' />}
                         </button>
+                        <button
+                            className='argo-button argo-button--base'
+                            style={{width: '100px'}}
+                            onClick={async () => {
+                                const downloadURL = services.applications.getDownloadLogsURL(
```

Comment:
Is there a possibility this request fails? If so could we have graceful handling of an error here?

Reply:
The `getDownloadLogsURL` just returns URL, so it cannot fail. However something might break during file generation. Changed API to print error into output, so user can see the error message in the file itself.